x/cert/client/cli/tx_cmd.go

```diff
@@ -0,0 +1,253 @@
+	if err := viper.BindPFlag(flagStart, cmd.Flags().Lookup(flagStart)); err != nil {
+		return err
+	}
+
+	cmd.Flags().Duration(flagValidTime, time.Hour*24*365, "certificate is not valid after this date. RFC3339")
+	if err := viper.BindPFlag(flagValidTime, cmd.Flags().Lookup(flagValidTime)); err != nil {
+		return err
+	}
+	cmd.Flags().Bool(flagOverwrite, false, "overwrite existing certificate if present")
+	if err := viper.BindPFlag(flagOverwrite, cmd.Flags().Lookup(flagOverwrite)); err != nil {
+		return err
+	}
+
+	flags.AddTxFlagsToCmd(cmd) // TODO - add just the keyring flags? not all the TX ones
+	return nil
+}
+
+func cmdGenerateClient() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                        "client",
+		Short:                      "",
+		SuggestionsMinimumDistance: 2,
+		RunE:                       doGenerateCmd,
+		SilenceUsage:               true,
+		Args:                       cobra.ExactArgs(0),
+	}
+	err := addGenerateflags(cmd)
+	if err != nil {
+		panic(err)
+	}
+
+	return cmd
+}
+
+func cmdGenerateServer() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                        "server",
+		Short:                      "",
+		SuggestionsMinimumDistance: 2,
+		RunE:                       doGenerateCmd,
+		SilenceUsage:               true,
+		Args:                       cobra.MinimumNArgs(1),
+	}
+	err := addGenerateflags(cmd)
+	if err != nil {
+		panic(err)
+	}
+
+	return cmd
+}
+
+func cmdPublish() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                        "publish",
+		Short:                      "",
+		SuggestionsMinimumDistance: 2,
+		RunE:                       sdkclient.ValidateCmd,
+	}
+
+	cmd.AddCommand(cmdPublishClient(),
+		cmdPublishServer())
+
+	return cmd
+}
+
+func cmdPublishClient() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                        "client",
+		Short:                      "",
+		SuggestionsMinimumDistance: 2,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return doPublishCmd(cmd)
+		},
+		SilenceUsage: true,
+		Args:         cobra.ExactArgs(0),
+	}
+	err := addPublishFlags(cmd)
+	if err != nil {
+		panic(err)
+	}
+
+	return cmd
+}
+
+func cmdPublishServer() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                        "server",
+		Short:                      "",
+		SuggestionsMinimumDistance: 2,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return doPublishCmd(cmd)
+		},
+		SilenceUsage: true,
+		Args:         cobra.ExactArgs(0),
+	}
+	err := addPublishFlags(cmd)
+	if err != nil {
+		panic(err)
+	}
+
+	return cmd
+}
+
+func addPublishFlags(cmd *cobra.Command) error {
+	cmd.Flags().Bool(flagToGenesis, false, "add to genesis")
+	if err := viper.BindPFlag(flagToGenesis, cmd.Flags().Lookup(flagToGenesis)); err != nil {
+		return err
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return nil
+}
+
+func addCertToGenesis(cmd *cobra.Command, cert types.GenesisCertificate) error {
```

Comment:
I still think that this shouldn't really be needed.

Reply:
It's used in the examples and is convenient whenever I'm setting up local test nets.

Reply:
it's just that we don't have it for anything else - providers, etc... are all created via the normal mechanism.  but ok, if it's useful then great.